### PR TITLE
Reload all sampler settings on change

### PR DESCRIPTION
**Pull request by martin308**
- don't store a default sampler to use, create the default for each dataset that uses the default
- remove reloading from each sampler and push this up into the collector so that we discard all of the samplers when the config changes so that they can be re-created with the new config data

### CODE REVIEW

File: collect/collect_test.go [hunk 1]
```diff
@@ -287,3 +289,69 @@ func TestDryRunMode(t *testing.T) {
...
+	}, conf.GetTraceTimeoutVal*2, conf.SendTickerVal)
+
+	conf.ReloadConfig()
+
+	assert.Eventually(t, func() bool {
+		_, ok := coll.datasetSamplers[dataset]
+		return !ok
+	}, conf.GetTraceTimeoutVal*2, conf.SendTickerVal)
+
+	span = &types.Span{
```
Comment (asdvalenzuela) [inline collect/collect_test.go:344]: What does adding this new span test that wasn't tested previously?

**Review by asdvalenzuela — APPROVED** [review]

File: collect/collect_test.go (hunk 1 above)
Comment (martin308) [reply collect/collect_test.go:344]: the new reload functionality, we first check that when a span is added that the map contains the dataset/sampler mapping, then we "reload" the configs that the map gets cleared and then when we send another span that the sampler is created again

Comment (asdvalenzuela) [reply collect/collect_test.go:344]: But you're sending two spans?

Comment (asdvalenzuela) [reply collect/collect_test.go:344]: Please ignore! I read it wrong.

**Pull request merged**